File: sgilt-core/src/main/java/net/franzka/sgilt/core/reservation/api/ReservationApi.java
package net.franzka.sgilt.core.reservation.api;

import net.franzka.sgilt.core.reservation.dto.ActiveReservationsDto;
import net.franzka.sgilt.core.reservation.dto.ProBoardCountsDto;
import net.franzka.sgilt.core.reservation.dto.ProReservationSummaryDto;
import net.franzka.sgilt.core.reservation.dto.ReservationMetaDto;
import net.franzka.sgilt.core.reservation.dto.ReservationSummaryDto;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;

import java.util.List;
import java.util.UUID;

@RequestMapping("api/v1/reservations")
public interface ReservationApi {

    @GetMapping
    ResponseEntity<List<ReservationSummaryDto>> getByEventId(@RequestParam UUID eventId);

    @GetMapping("/active")
    ResponseEntity<ActiveReservationsDto> getActive();

    @GetMapping("/pro")
    ResponseEntity<List<ProReservationSummaryDto>> getProReservations();

    @GetMapping("/pro/counts")
    ResponseEntity<ProBoardCountsDto> getProBoardCounts();

    @GetMapping("/{reservationId}")
    ResponseEntity<ReservationMetaDto> getDetail(@PathVariable UUID reservationId);

    @PostMapping("/{reservationId}/cancel")
    ResponseEntity<Void> cancel(@PathVariable UUID reservationId);
}

File: sgilt-front/app/data/reservation/api/reservationApi.ts
import { apiFetch } from '~/composables/useApi'
import type { ActiveReservationsDto } from '../dto/ActiveReservationsDto'
import type { ProBoardCountsDto } from '../dto/ProBoardCountsDto'
import type { ProReservationSummaryDto } from '../dto/ProReservationSummaryDto'
import type { ReservationMetaDto } from '../dto/ReservationMetaDto'
import type { EventReservationSummaryDto } from '~/data/evenement/dto/EventReservationSummaryDto'

export async function getReservationsByEventApi(eventId: string): Promise<EventReservationSummaryDto[]> {
  return apiFetch<EventReservationSummaryDto[]>(`/reservations`, { params: { eventId } })
}

export async function getReservationMetaApi(reservationId: string): Promise<ReservationMetaDto> {
  return apiFetch<ReservationMetaDto>(`/reservations/${reservationId}`)
}

export async function cancelReservationApi(reservationId: string): Promise<void> {
  await apiFetch(`/reservations/${reservationId}/cancel`, { method: 'POST' })
}

export async function getActiveReservationsApi(): Promise<ActiveReservationsDto> {
  return apiFetch<ActiveReservationsDto>('/reservations/active')
}

export async function getProReservationsApi(): Promise<ProReservationSummaryDto[]> {
  return apiFetch<ProReservationSummaryDto[]>('/reservations/pro')
}

export async function getProBoardCountsApi(): Promise<ProBoardCountsDto> {
  return apiFetch<ProBoardCountsDto>('/reservations/pro/counts')
}
